components/zoho_crm/sources/common/http-based/base.js

```diff
@@ -0,0 +1,280 @@
+        .map(this.zoho_crm.genericApiGetCall);
+      const responses = await Promise.all(apiCalls);
+      return responses.map(({ data }) => data);
+    },
+    /**
+     * This function returns the list of webhook events that the component
+     * listens to. This corresponds to the `events` parameter that must be
+     * specified when creating a webhook in Zoho CRM, as per the docs:
+     * https://www.zoho.com/crm/developer/docs/api/v2/notifications/enable.html
+     *
+     * The function must be customised accordingly by each component based on
+     * the specific event source requirements.
+     *
+     * For example, an event source that listens to new or updated _solutions_
+     * would look like this:
+     * @example
+     * function getEvents() {
+     *   return [
+     *     "Solutions.create",
+     *     "Solutions.edit",
+     *   ];
+     * }
+     *
+     * @returns {string[]} The list of webhook events that this event source
+     * will listen to.
+     */
+    getEvents() {
+      throw new Error("getEvents is not implemented");
+    },
+    /**
+     * This function provides information about the types of CRUD operations
+     * that the event source listens to.
+     *
+     * For example, an event source that reacts to the creation or edition of a
+     * resource would return a list of size 2, with each entry corresponding to
+     * each of the event types previously mentioned.
+     *
+     * The type of each entry in the returned list must be an object containing
+     * the following attributes:
+     * - `op`: The operation code. Acceptable values are `create`, `delete` or
+     *   `edit`.
+     * - `flagName`: The name of the attribute in a Zoho CRM's module object
+     *   that signals whether the operation is supported. Acceptable values are
+     *   `creatable`, `deletable` and `editable`, for creation, deletion and
+     *   edition operations, respectively.
+     * - `labelPostfix`: A user-friendly description of the operation.
+     *
+     * For convenience, the `crud-operations` module exports functions that
+     * provide object for CRUD operations, so for example an event source that
+     * reacts to creation or edition events can implement this function in the
+     * following way:
+     * @example
+     * const { createOpData, editOpData } = require("./crud-operations");
+     * function getEvents() {
+     *   return [
+     *     createOpData(),
+     *     editOpData(),
+     *   ];
+     * }
+     *
+     * @returns {object[]} The list of CRUD operations that this event source
+     * reacts to
+     */
+    getSupportedOps() {
+      throw new Error("getSupportedOps is not implemented");
+    },
+    generateMeta({
+      event, resource,
+    }) {
+      const {
+        headers: { "x-amzn-trace-id": eventId },
+        body: {
+          module, operation,
+        },
+      } = event;
+      const {
+        id: resourceId,
+        Created_Time: creationTime,
```

Comment:
Are these timestamps reliably present in all cases?

Reply:
Yes, even when a record is newly created. In such cases, `Modified_Time` equals `Created_Time`.